tests/test_helpers.py

```diff
@@ -0,0 +1,33 @@
+from django.contrib.contenttypes.models import ContentType
+
+from djangocms_versioning.constants import DRAFT, PUBLISHED
+from djangocms_versioning.models import Version
+from filer.models import File, Folder
+
+from .base import BaseFilerVersioningTestCase
+from djangocms_versioning_filer import helpers
+
+
+class HelperTests(BaseFilerVersioningTestCase):
+
+    def test_check_folder_exists_in_folder(self):
+        self.assertFalse(helpers.check_folder_exists_in_folder(self.folder, 'olivertwist'))
+        subfolder = Folder(
+            parent=self.folder,
+            name='olivertwist'
+        )
+        subfolder.save()
+        self.assertEquals(helpers.check_folder_exists_in_folder(self.folder, 'olivertwist'), subfolder)
+
+    def test_add_subfolder(self):
```

Comment:
This test seems to be about 4 tests in actuality. IMO this should be separated out for easier maintainability and readability (nice test names or failing that docstrings for exactly what you're testing would help).

Reply:
I don't see how it is four tests. It's a single test with several assertions that all together ensure that the function does what it should.

Reply:
Yep, which is against unit test best practices. I've know devs who go as extreme as "one assert per unit test, no more". I'm not _that_ extreme, but from experience of having to maintain python code and python tests for years, it is much easier when you do one unit test per case rather than per function.
A good design pattern to follow is AAA: http://wiki.c2.com/?ArrangeActAssert
If you can't put a test into that pattern then it probably means you need to break it down. I would really recommend following AAA whenever possible.

That said, if you follow my advice above you won't even need this unit test because you will have removed `add_subfolder` all together :wink: